### ISSUE TITLE
Use WebExtensions API

### ISSUE THREAD

**Issue by Gitoffthelawn** [muzuiget/referrer_control#58]
As of Firefox 57, only WebExtensions API will be allowed.

**Comment by yfdyh000** [comment]
https://github.com/muzuiget/referrer_control
> Note: Because Firefox 57 will drop support legacy add-ons, and I don't had any time to migrate it to WebExtensions yet, this project is in maintain mode, bugfix only, no new feature.

Alternatives:
https://addons.mozilla.org/firefox/addon/header-editor/
https://addons.mozilla.org/firefox/addon/referercontrol/

**Comment by Gitoffthelawn** [comment]
Thanks for the references to alternatives.

Not to pester... do you think you will have any time to migrate this extension to the WebExtensions API, or is it at EOL?

**Comment by yfdyh000** [comment]
I don't know, I did not use it.

**Comment by IzzySoft** [comment]
Note: the [addons.mozilla.org/firefox/addon/referercontrol](https://addons.mozilla.org/firefox/addon/referercontrol/) addon has some restrictions compared to this here:

* no "same domain" policy
* no "fromSite › toSite" mappings
* uses 3rd party Javascript plus Disqus within an iFrame on its config page (can be blocked by NoScript)

And before I am asked: As much as I'd love to help out, I'm no FF dev.

**Comment by IzzySoft** [comment]
Closed without comment must mean "dead end, won't fix" – 'cause were it to be fixed "at some point in the future", the issue'd been kept open.

So one more victim to Mozilla's policies :frowning_face:

**Comment by muzuiget** [comment]
Yes. Because Github has release a "Archive Repository" feature. So closing some issues which I don't want to finish for a long time.

**Comment by IzzySoft** [comment]
@muzuiget thanks for explaining! Need to look what that "Archive Repository" feature is about, might need that for some of my projects as well.

And just to clarify: I don't blame you for this, you've got my full understanding and sympathy. I rather blame Mozilla for cutting off compatibility even before they have the "replacement API" fully available. There're many addons which simply cannot (yet) be converted into WE as Mozilla hasn't yet provided corresponding APIs.

Your Referrer_Control addon is one of my essentials. Two others are blocked by described API problem. Looks like I have to freeze FF on v55; being forced to completely start fresh I'm rather considering switching to a different browser …

**Comment by Gitoffthelawn** [comment]
@IzzySoft Although maybe not as wonderful as this extension, there are several WebExtension API compatible referrer (referer) extensions already available.

**Comment by IzzySoft** [comment]
@Gitoffthelawn I've seen none suitable yet. The one I checked failed privacy-wise (see above).

As I did with FF3.6 and 27, guess I'll again "freeze" it at its current version and wait half a year to see what happens. I'm fed up with their politics. If I'd knew a suitable alternative to the browser, I'd consider switching that.

**Comment by Gitoffthelawn** [comment]
@IzzySoft Try Smart Referer.  The source code is here on GitHub: https://github.com/meh/smart-referer

If you have any issues with it, you can post them in that repo.
